src/Cthulhu.jl

```diff
@@ -845,57 +842,55 @@ end
 
 function do_typeinf!(interp::AbstractInterpreter, mi::MethodInstance)
     result = InferenceResult(mi)
-    @static if isdefined(CC, :engine_reserve)
-        ci = CC.engine_reserve(interp, mi)
-        result.ci = ci
-    end
+    ci = CC.engine_reserve(interp, mi)
```

Comment:
This PR requires us to bump the minimum compatible Julia version to 1.11, requiring Cthulhu's minor version to be bumped jointly.

Reply:
Oh yes, I think these line edits were made in your original PR. Do you think that this PR is fundamentally incompatible with v1.11 (e.g. because code instances are not available as much), or can we do with keeping support for v1.11 while throwing out the `MethodInstance`-based caching?

Reply:
Feel free to create the branch with a name of your choosing, and I'll redirect the PR to it.

Reply:
ok, I created one named `2.17`.